### ISSUE TITLE
Does not run on python 3.9: TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'

### ISSUE THREAD

**Issue by jonorlin** [RhetTbull/osxphotos#1708]
**Describe the bug**
osxphotos was running just fine before. But something changed and i can't get it to run.

**To Reproduce**
after I run pipx reinstall-all i get this

```
jon@Jons-Mac-Studio venvs % pipx reinstall-all
uninstalled osxphotos! ✨ 🌟 ✨
⚠️  Note: osxphotos was already on your PATH at /Users/jonorlin/.pyenv/shims/osxphotos
  installed package osxphotos 0.68.6, installed using Python 3.12.6
  These apps are now globally available
    - osxphotos
done! ✨ 🌟 ✨
```

But then if i try to run osxphotos or even just check the version, i get this

```
jon@Jons-Mac-Studio venvs % osxphotos -v
Traceback (most recent call last):
  File "/Users/jon/.pyenv/versions/3.9.2/bin/osxphotos", line 5, in <module>
    from osxphotos.__main__ import cli_main
  File "/Users/jon/.pyenv/versions/3.9.2/lib/python3.9/site-packages/osxphotos/__init__.py", line 17, in <module>
    from .iphoto import (
  File "/Users/jon/.pyenv/versions/3.9.2/lib/python3.9/site-packages/osxphotos/iphoto.py", line 64, in <module>
    from .photoinfo import PhotoInfo
  File "/Users/jon/.pyenv/versions/3.9.2/lib/python3.9/site-packages/osxphotos/photoinfo.py", line 56, in <module>
    from .bookmark import resolve_bookmark_path
  File "/Users/jon/.pyenv/versions/3.9.2/lib/python3.9/site-packages/osxphotos/bookmark.py", line 12, in <module>
    def resolve_bookmark_path(bookmark_data: bytes) -> pathlib.Path | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

**Expected behavior**
I expect to run osxphotos commands from the command line and script but get this error.

**Desktop (please complete the following information):**
 - OS: 14.6.1
 - osxphotos version (`osxphotos --version`) 0.68.6

Note i also get pipx 1.7.1 is already installed and up-to-date
and Homebrew is up-to-date.

Thanks for the great program. Hopefully this is something simple to fix.

**Comment by oPromessa** [comment]
Just a small note. 

- pipx reinstall-all mentions Python 3.12.6
- osxphotos -v mentions `jon/.pyenv/versions/3.9.2/bin/osxphotos`

Maybe double check if you are running the correct python version and venv where you've installed it?

**Comment by oPromessa** [comment]
@jonorlin yes I think it's probably that. This construct is only available on python 3.10 and above version and from your log it seems you're running python 3.9

Really hope it helps. 

[TypeError: unsupported operand type(s) for |:](https://stackoverflow.com/questions/76712720/typeerror-unsupported-operand-types-for-type-and-nonetype)

**Comment by RhetTbull** [comment]
@jonorlin You have two versions installed as @oPromessa said. One appears to be installed in a virtual env with Python 3.9.

There is a bug in OSXPhotos that does not work on 3.9 as @oPromessa noted. I will fix this as soon as I'm able but please note that  I will soon drop support for Python 3.9 once I've ported OSXPhotos to 3.13. OSXPhotos has the goal of supporting the most current 3 releases of Python (now 3.10, 3.11, 3.13) but I've not yet had time to port to 3.13.

In the meantime, you can fix this by deleting the 3.9 version from your path. (`rm /Users/jonorlin/.pyenv/shims/osxphotos`) or by using Python 3.10 or 3.11.

**Comment by jonorlin** [comment]
Thanks @RhetTbull and @oPromessa for your speedy replies.
You both correctly identified the problem. I had multiple version of Python and my zshrc shell script was pointing to the old Python 3.9 version. Pointing it to the newer version solved the problem.

When i ran the python update, it updated to 3.13. So we i first tried OSXPhotos, i ran it under 3.13 and it seemed to work. But i realize it is not ported yet, so i'll downgrade to 3.12.

In the last comment, I think there is a typo here:
the most current 3 releases of Python (now 3.10, 3.11, 3.13...
that 3.13 should be 3.12.

Much thanks!
